mobile/src/libs/components/input/input.tsx

```diff
@@ -25,26 +26,46 @@ const Input = <T extends FieldValues>({
   errors,
   label,
   name,
+  secure,
   placeholder,
 }: Properties<T>): JSX.Element => {
+  const [isFocused, setIsFocused] = useState(false);
   const { field } = useFormController({ name, control });
 
   const { value, onChange, onBlur } = field;
 
   const error = errors[name]?.message;
   const hasError = Boolean(error);
 
+  const isFieldFilled = Boolean(value);
+
+  const handleFocus = (): void => {
+    setIsFocused(true);
+  };
+
+  const handeBlur = (): void => {
+    setIsFocused(false);
+    onBlur();
+  };
+
   return (
     <View>
-      <Text>{label}</Text>
+      <Text style={styles.label}>{label}</Text>
       <TextInput
         onChangeText={onChange}
         value={value}
-        onBlur={onBlur}
+        onFocus={handleFocus}
+        onBlur={handeBlur}
+        secureTextEntry={secure}
         placeholder={placeholder}
-        style={styles.input}
+        style={[
+          styles.input,
+          !hasError && (isFocused || isFieldFilled) && styles.filledInput,
```

Comment:
let's not apply this styles if input is not in focus, it will provide better user experience
```suggestion
         !hasError && isFocused && styles.focusedInput,
```